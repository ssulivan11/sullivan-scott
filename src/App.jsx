import React, { useState, useEffect } from 'react'
import { hot } from 'react-hot-loader/root'
import throttle from 'lodash.throttle'

import content from './helpers/content'

import Home from './containers/home/Home'
import About from './containers/about/About'
import Contact from './containers/contact/Contact'
import style from './scss/main.scss'

require('./assets/favicon.ico')

const App = () => {
  const getWindowSize = () => ({
    height: window.innerHeight,
    width: window.innerWidth,
  })
  const getScrollPosition = () => ({ y: window.pageYOffset })

  const [windowSize, setWindowSize] = useState(getWindowSize())
  const [scrollPosition, setScrollPosition] = useState(getScrollPosition())

  const handleResize = throttle(() => {
    setWindowSize(getWindowSize())
  }, 50)
  const handleScroll = throttle(() => {
    setScrollPosition(getScrollPosition())
  }, 50)

  useEffect(() => {
    window.addEventListener('resize', handleResize)
    window.addEventListener('scroll', handleScroll)
    return () => {
      window.removeEventListener('resize', handleResize)
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  const isHomeActive = windowSize.height > scrollPosition.y

  const { mainHeadingText, subHeadingText, social } = content

  return (
    <div className={style.container}>
      <Home
        isActive={isHomeActive}
        windowSize={windowSize}
        mainHeadingText={mainHeadingText}
        subHeadingText={subHeadingText}
        data-test='home'
      />
      <About isActive={!isHomeActive} windowSize={windowSize} data-test='about' />
      <Contact isActive={!isHomeActive} windowSize={windowSize} social={social} data-test='contact' />

      <div className={style['gradient-background']} style={{ height: windowSize.height }} />
    </div>
  )
}

export default hot(App)
