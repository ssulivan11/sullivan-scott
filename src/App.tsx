import React, { useEffect, useState, StrictMode } from 'react'
import { hot } from 'react-hot-loader/root'
import throttle from 'lodash.throttle'

import content from './helpers/content'
import ErrorBoundary from './components/ErrorBoundary/ErrorBoundary'

import Home from './containers/home/Home'
import About from './containers/about/About'
import Contact from './containers/contact/Contact'
import * as style from './scss/main.scss'

interface AppProps {
  compiler: string
  framework: string
}

const App: React.FunctionComponent<AppProps> = () => {
  const getWindowSize = () => ({
    height: document.documentElement.clientHeight,
    width: document.documentElement.clientWidth,
  })
  const getScrollPosition = () => ({ y: window.pageYOffset })

  const [windowSize, setWindowSize] = useState(getWindowSize())
  const [scrollPosition, setScrollPosition] = useState(getScrollPosition())

  const handleResize = throttle(() => {
    setWindowSize(getWindowSize())
  }, 50)
  const handleScroll = throttle(() => {
    setScrollPosition(getScrollPosition())
  }, 50)

  useEffect(() => {
    window.scrollTo(0, 0)
    window.addEventListener('resize', handleResize)
    window.addEventListener('scroll', handleScroll)
    return () => {
      window.removeEventListener('resize', handleResize)
      window.removeEventListener('scroll', handleScroll)
    }
  }, [])

  const heightWithGive = windowSize.height - (windowSize.height / 100) * 20
  const active =
    heightWithGive > scrollPosition.y ? 'home' : heightWithGive * 2 > scrollPosition.y ? 'about' : 'contact'

  const { mainHeadingText, subHeadingText, social, skills, bioHeading, bioText, bioImg } = content

  return (
    <div className={style.container}>
      <StrictMode>
        <ErrorBoundary>
          <Home
            active={active}
            windowSize={windowSize}
            mainHeadingText={mainHeadingText}
            subHeadingText={subHeadingText}
            data-test='home'
          />
        </ErrorBoundary>
        <ErrorBoundary>
          <About
            active={active}
            windowSize={windowSize}
            bioHeading={bioHeading}
            bioText={bioText}
            skills={skills}
            bioImg={bioImg}
            data-test='about'
          />
        </ErrorBoundary>
        <ErrorBoundary>
          <Contact active={active} windowSize={windowSize} social={social} data-test='contact' />
        </ErrorBoundary>
      </StrictMode>
    </div>
  )
}

export default hot(App)
