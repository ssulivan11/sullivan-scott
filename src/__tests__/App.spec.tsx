import * as React from 'react'
import * as ReactTestUtils from 'react-dom/test-utils'
import { shallow, mount } from 'enzyme'

import App from '../App'

window.scrollTo = jest.fn()

const defaultProps = { compiler: 'TypeScript', framework: 'React' }

const setup = (renderer = mount) => {
  let component

  ReactTestUtils.act(() => {
    component = renderer(<App {...defaultProps} />)
  })

  return {
    getInstance: () => component,
    getApp: () => component.find(App),
    debug: () => component.debug(),
  }
}

let wrapper
describe('<App />', () => {
  it('active container', () => {
    wrapper = setup(shallow)

    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })
})

describe('<App />', () => {
  it('active container "contact"', () => {
    wrapper = setup(mount)

    ReactTestUtils.act(() => {
      // @ts-ignore
      window.dispatchEvent(new window.UIEvent('scroll'))
      // @ts-ignore
      window.dispatchEvent(new window.UIEvent('resize'))
    })

    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('active container "home"', () => {
    wrapper = setup(mount)

    // @ts-ignore
    Object.defineProperty(document.documentElement, 'clientHeight', { writable: true, configurable: true, value: 1000 })
    // @ts-ignore
    window.pageYOffset = 500

    ReactTestUtils.act(() => {
      // @ts-ignore
      window.dispatchEvent(new window.UIEvent('scroll'))
      // @ts-ignore
      window.dispatchEvent(new window.UIEvent('resize'))
    })

    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('active container "about"', () => {
    wrapper = setup(mount)

    // @ts-ignore
    Object.defineProperty(document.documentElement, 'clientHeight', { writable: true, configurable: true, value: 600 })
    // @ts-ignore
    window.pageYOffset = 500

    ReactTestUtils.act(() => {
      // @ts-ignore
      window.dispatchEvent(new window.UIEvent('scroll'))
      // @ts-ignore
      window.dispatchEvent(new window.UIEvent('resize'))
    })

    expect(wrapper).toBeDefined()
    expect(wrapper).toMatchSnapshot()
  })

  it('should unnmount resize listeners functions', () => {
    const wrapper = mount(<App {...defaultProps} />)
    wrapper.unmount()
  })
})
