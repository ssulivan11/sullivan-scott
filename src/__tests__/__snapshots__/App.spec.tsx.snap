// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> active container "about" 1`] = `
Object {
  "debug": [Function],
  "getApp": [Function],
  "getInstance": [Function],
}
`;

exports[`<App /> active container "contact" 1`] = `
Object {
  "debug": [Function],
  "getApp": [Function],
  "getInstance": [Function],
}
`;

exports[`<App /> active container "home" 1`] = `
Object {
  "debug": [Function],
  "getApp": [Function],
  "getInstance": [Function],
}
`;

exports[`<App /> active container 1`] = `
Object {
  "debug": [Function],
  "getApp": [Function],
  "getInstance": [Function],
}
`;
